GoedeDoelenHelpen/Data/ApplicationDbContext.cs

```diff
@@ -1,16 +1,190 @@
             : base(options)
         {
         }
     }
+
+    public class ApplicationUser: IdentityUser
+    {
+
+        public IEnumerable<Donation> Donations { get; set; }
+    }
+
+    public class Charity
+    {
+        public int Id { get; set; }
+        public string Adress { get; set; }
+        public string Name { get; set; }
+        public string KVKNumber { get; set; }
+        public string FiscalNumber { get; set; }
+        public string IBAN { get; set; }
+
+        /// <summary>
+        /// relation to intersention entity CharityApplicationUser
+        /// </summary>
+        public IEnumerable<CharityApplicationUser> CharityApplicationUsers { get; set; }
+
+        public IEnumerable<CharityDonation> Donations { get; set; }
+
+        public IEnumerable<CharityEvent> Events { get; set; }
+    }
+
+    public class CharityApplicationUser
+    {
+        public int Id { get; set; }
+
+        public string ApplicationUserId { get; set; }
+        public ApplicationUser ApplicationUser { get; set; }
+
+        public int CharityId { get; set; }
+        public Charity Charity { get; set; }
+
+        public CharityApplicationUserRole CharityApplicationUserRole { get; set; }
+    }
+
+    public enum CharityApplicationUserRole
+    {
+        Admin,
+        User
+    }
+
+    public class Project
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public int CharityId { get; set; }
+        public Charity Charity { get; set; }
+
+        public IEnumerable<ProjectDonation> Donations { get; set; }
+
+        public IEnumerable<ProjectEvent> Events { get; set; }
+    }
+
+
+    /// <summary>
+    /// base type for all the different types of Donations
+    /// </summary>
+    /// <remarks>
+    /// EF will putt all different donation in one table and usses a shadow propperty the "discriminator" to determin wich type it actualy is.
+    /// </remarks>
+    public abstract class Donation
+    {
+        public int Id { get; set; }
+        public decimal Amount { get; set; }
+        public DateTime DateTime { get; set; }
+
+        public string ApplicationUserId { get; set; }
+        public ApplicationUser ApplicationUser { get; set; }
+    }
+
+    public class CharityDonation: Donation
+    {
+        public int CharityId { get; set; }
+        public Charity Charity { get; set; }
+    }
+
+    public class ProjectDonation: Donation
+    {
+        public int ProjectId { get; set; }
+        public Project Project { get; set; }
+    }
+
+
+    public abstract class Event
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+    }
+
+    public class ProjectEvent: Event
+    {
+        public int ProjectId { get; set; }
+        public Project Project { get; set; }
+    }
+
+    public class CharityEvent: Event
+    {
+        public int CharityId { get; set; }
+        public Charity Charity { get; set; }
+    }
+
+    public class EventTeam
```

Comment:
EventTeam moet volgens de ERD een relatie met ApplicationUser hebben, deze bestaat echter niet in de code